src/t_zset.c

```diff
@@ -1165,7 +1166,43 @@ unsigned long zsetLength(const robj *zobj) {
     return length;
 }
 
+/* Factory method to return a zset.
+ *
+ * The size hint indicates approximately how many items will be added,
+ * and the value len hint indicates the approximate individual size of the added elements,
+ * they are used to determine the initial representation. */
```

Comment:
maybe add a comment that if the hints are not known, and underestimation or 0 is suitable.